components/Tabs/Tabs.js

```diff
@@ -1,33 +1,33 @@
 class TabLink {
   constructor(tabElement){
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
+    this.tabElement = tabElement;
     
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
+    this.tabData = this.tabElement.dataset.tab; 
     
     // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
     
-    /* <- Delete this comment block when you work on the if statement
     // Check to see if this.tabData is equal to 'all'
-    if(){
+    if(this.tabData === 'all'){
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = [...document.body.querySelectorAll('.cards-container .card')];
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
+      this.cards = [...document.body.querySelectorAll('.cards-container .card')]
```

Comment:
Alternatively, you could have used ``` this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);``` here  and then in line 22 you would have ```this.cards = Array.from(this.cards).map(element => new TabCard(element));```. Your approach works too.